### ISSUE TITLE
DO cluster fails to come up in rancher server running in a proxy environment

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#20903]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
1. Set up a proxy environment.
2. Run the rancher server and pass HTTP and HTTPS parameters while doing the docker run.

**Actual Result:**
DO cluster fails to come up whereas an AWS cluster comes up.
<img width="1286" alt="Screen Shot 2019-06-07 at 7 12 20 PM" src="https://user-images.githubusercontent.com/26032343/59547341-4e654400-8ef2-11e9-8f16-28e504882add.png">

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): rancher: v2.2.5-rc1
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): DO

**Comment by cjellick** [comment]
I do not think we have enough detail to repro. Can you go into more detail on "set up a proxy environment"

**Comment by sowmyav27** [comment]
The rancher server was started using this command
```
docker run -d --restart=unless-stopped \
  -p 80:80 -p 443:443 \
  -v /home/ubuntu/state:/var/lib/rancher \
  -e HTTP_PROXY="http://<proxy-ip>:8888" \
  -e HTTPS_PROXY="http://<proxy-ip>:8888" \
  -e NO_PROXY="localhost,127.0.0.1,0.0.0.0,<subnet>/16" \
  rancher/rancher:<rancher version>
```

**Comment by cjellick** [comment]
but what was the proxy you setup?
